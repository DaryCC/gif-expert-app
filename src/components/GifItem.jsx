import PropTypes from 'prop-types';
import React from 'react'

const GifItem = ({url, title,id}) => {

  // console.log(props);
  return (
    <div className="card">
      <img src={url} alt={title}/>
      <p>{title}</p>
    </div>
  )
}

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
export default GifItem

