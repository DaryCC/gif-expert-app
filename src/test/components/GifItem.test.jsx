import React from 'react';

import {render,screen} from '@testing-library/react';

import GifItem from '../../components/GifItem';



describe ('Pruebas en GifItem',()=>{

  const title = 'Dary';
  const url = 'https://www.primevideo.com/';

  test('GifGrid debe retornar title y url',()=>{

    const {container}= render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  })


  test('debe de mostrar la imagen con el URL y el ALT indicado', () => {
    render(<GifItem title={title} url={url} />);
    // screen.debug();
    console.log(screen.getByRole('img').src);
    // expect(screen.getByRole('img').alt).toBe(title);
    // expect(screen.getByRole('img').src).toBe(url);
    const {src,alt}= screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);


  })

  test('should mostrar el {titulo en el body}', () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  })

}

);
