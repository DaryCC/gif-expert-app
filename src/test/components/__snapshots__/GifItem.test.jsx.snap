// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem GifGrid debe retornar title y url 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dary"
      src="https://www.primevideo.com/"
    />
    <p>
      Dary
    </p>
  </div>
</div>
`;
